File: Daily Flash/week 4/day 1/18-DailyFlash_Solutions/3-Feb_Solutions_Three/Python/prog2.py
def bin_hex(b):
    hex_d = {10 : 'A', 11 : 'B', 12 : 'C', 13 : 'D', 14 : 'E', 15 : 'F'}
    rem = 0
    i = 0
    dec = 0

    j = 0
    rem1 = 0
    div = 0
    while(b != 0):
        rem = b % 10
        if(not (rem == 0 or rem == 1)):
            print("Enter valid bin")
            exit(1)
        dec = dec + rem * (2 ** i)
        i += 1
        b = b / 10
        b = int(b)
    print("Dec : ", dec)
    print("Hex : ", end = "")
    while(j >= 0):
        div = dec / 16 ** j
        div = int(div)
        rem1 = dec % 16 ** j
        if(div >= 0 and div < 16):
            if(div < 10):
                print(div, end = "")
            else:
                print(hex_d[div], end = "")
            j -= 1
            dec = rem1
        else:
            j += 1

b = int(input("num : "))
bin_hex(b)

File: Daily Flash/week 4/day 1/18-DailyFlash_Solutions/3-Feb_Solutions_Three/Java/prog2.java
import java.util.Scanner;

class Demo {
	public static void printHex(int num) {	
		switch(num) {
			case 10:
				System.out.print("A");
				break;
			case 11:
				System.out.print("B");
				break;
			case 12:
				System.out.print("C");
				break;
			case 13:
				System.out.print("D");
				break;
			case 14:
				System.out.print("E");
				break;
			case 15:
				System.out.print("F");
				break;
			default:
				System.out.print(num);
				break;
		}
	}
	public static void bin_hex(long bin) {
		int dec = 0, rem, i = 0;
		int temp, rem1, j = 0, div;
		while(bin != 0) {
			rem = (int)(bin % 10);
			if(!(rem == 0 || rem == 1)) {
				System.out.println("Enter valid bin.");
				System.exit(1);
			}
			dec += rem * Math.pow(2, i);
			i++;
			bin /= 10;
		}
		System.out.print("Hex : ");
		temp = dec;
		while(j >= 0) {
			div = temp /(int)Math.pow(16, j);
			rem1 = temp % (int)Math.pow(16, j);
			if(div >= 0 && div < 16) {
				printHex(div);
				j--;
				temp = rem1;
			} else {
				j++;
			}
		}

	}
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter bin : ");
		long bin = sc.nextLong();
		bin_hex(bin);
	}
}
